### ISSUE TITLE
[rush] "rush build" always rebuilds a project with an empty build script

### ISSUE THREAD

**Issue by dzearing** [microsoft/web-build-tools#326]
Repro:

Have 2 or more packages in a repo that depend on a core package.
In the core package, have an empty string for clean/build scripts:

```json
{
  "scripts": {
    "clean": "",
    "build": ""
  }
}
```

Run `rush build`
Run `rush build` again

Expected:
Everything is skipped.

Resulted:
Everything is built again, because an empty "build" script bypasses the build task completely and invalidates downstream dependencies.

**Comment by nickpape-msft** [comment]
This is a legitimate issue, thanks David.

**Comment by gregjacobs** [comment]
Just ran into this issue as well. We have a 'common' package that simply has some JavaScript files in it that don't need to be "built" so-to-speak, but other packages rely on 'common' as a dependency. When files in the 'common' package change, their dependents should be rebuilt.

**Comment by octogonz** [comment]
There seem to be two different issues being reported here:

1. @dzearing expected  `rush build` to SKIP when invoked a second time (whereas he reported that it incorrectly rebuilds).  I am unable to repro this.  What I tried was modifying [my-controls/package.json](https://github.com/microsoft/rush-example/blob/master/libraries/my-controls/package.json) in the **rush-example** repo to have the empty `"scripts"` section as described above.  When I ran `rush build` the second time, all projects were skipped as expected.  Whereas e.g. if I modified **my-toolchain**, then `rush build` correctly rebuilt that project and also **my-app**, while skipping **my-controls** as expected.

2. @gregjacobs expected `rush build` to REBUILD when invoked a second time after the scriptless project was modified (whereas he reported that it incorrectly skips). I was able to repro this easily.  It seems like wrong behavior, although I suspect the original design didn't think about this situation at all.

I'm going to move #2 to a separate issue.  We should close #1 after a week if nobody can provide a repo.
